### ISSUE TITLE
A newline rule and its space twin both write the run they share, so `--fix` exits clean over a file one of them still refuses

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#704]
### What minimal example or steps are needed to reproduce the bug?

A newline rule and the space rule of the same subject write one and the same run, and their `always` options ask for different things in it. Under `--fix` both warnings are raised and both are marked fixed, so the run ends reporting nothing at all — over a file the configuration still refuses:

```css
/* input */
a, b { c: d }
```

```shell
# `selector-list-comma-newline-after: "always"` listed first, then `selector-list-comma-space-after: "always"`
--fix leaves `a, b { c: d }` and reports nothing, exit 0
a check over that output says 1:2 Expected newline after ","

# the same two rules the other way round
--fix leaves `a,\n b { c: d }` and reports nothing, exit 0
a check over that output says 1:2 Expected single space after ","
```

So `--fix` exits clean, the next `--fix` exits clean again, and only a run without `--fix` says the file is wrong. Which of the two spellings survives is decided by the configuration order.

Measured over every twin pair of the registry under every primary option each takes, across nine shapes and both configuration orders — 4716 rows. **424 rows fall into this class, over 66 option pairs**, across fourteen newline rules, each against its space twin: `block-closing-brace-newline-before`, `block-opening-brace-newline-after`, `block-opening-brace-newline-before`, `declaration-block-semicolon-newline-after`, `declaration-block-semicolon-newline-before`, `function-comma-newline-after`, `function-comma-newline-before`, `function-parentheses-newline-inside`, `media-query-list-comma-newline-after`, `selector-list-comma-newline-after`, `selector-list-comma-newline-before`, `value-list-comma-newline-after`, `value-slash-newline-after` and `value-slash-newline-before`.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/selector-list-comma-newline-after": "always",
    "@stylistic/selector-list-comma-space-after": "always"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

Either the run says what is still wrong with the file it leaves, or the pair agrees on one spelling and leaves nothing to say. A `--fix` that exits clean over a file the same configuration rejects is the worst of the two.

### What actually happened?

Both rules write, the later-listed one wins, and both warnings are dropped as fixed. The output violates whichever rule wrote first, and nothing reports it until somebody runs Stylelint without `--fix`.

### Do you have a proposal to fix the bug?

The four `declaration-colon-*` and `declaration-block-semicolon-*` rules already answer this for the run they share, through `writesSharedRun` and `whitespaceAsked`: one rule owns the run, writes it, and the rule that would have written it differently reports and leaves it alone. The twins above share a run in the same way and read none of it.

Found while adding a fixer to `at-rule-name-newline-after` (see [#696](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/696)), which makes that rule the 67th pair: the same measurement over the branch gives 430 rows and 67 pairs, the one added being `at-rule-name-newline-after "always"` against `at-rule-name-space-after "always"`.
